library/std/src/io/stdio.rs

```diff
@@ -120,6 +120,14 @@ impl Read for StdinRaw {
     fn read_to_string(&mut self, buf: &mut String) -> io::Result<usize> {
         handle_ebadf(self.0.read_to_string(buf), 0)
     }
+
+    fn read_exact(&mut self, buf: &mut [u8]) -> io::Result<()> {
+        handle_ebadf(self.0.read_exact(buf), ())
+    }
```

Comment:
This is incorrect, this should return `UnexpectedEof` instead (which the default implementation does), as the buffer hasn't been filled.

Reply:
Ah, good catch. I missed that ebadf is interpreted differently for reading vs writing. Should be fixed now.